webrender/src/tiling.rs

```diff
@@ -2594,18 +2592,16 @@ impl FrameBuilder {
                         continue;
                     }
 
-                    match layer.composite_kind {
-                        CompositeKind::None => {}
-                        CompositeKind::Simple(..) | CompositeKind::Complex(..) => {
-                            let layer_rect = layer.xf_rect.as_ref().unwrap().bounding_rect;
-                            let location = RenderTaskLocation::Dynamic(None, layer_rect.size);
-                            let new_task = RenderTask::new_alpha_batch(next_task_index,
-                                                                       layer_rect.origin,
-                                                                       location);
-                            next_task_index.0 += 1;
-                            let prev_task = mem::replace(&mut current_task, new_task);
-                            alpha_task_stack.push(prev_task);
-                        }
+                    let composite_count = layer.composite_ops.count();
```

Comment:
why do you need to separate task creation from task initialization (which is happening in `PopStackingContext`)? Could we have all the logic in one place (in the pop code) instead?